### PR TITLE
Opening time offset field visibility fixed on details pane

### PR DESCRIPTION
**Pull request by anton-plotnikov**
<h1>Opening time offset field behavior fixed</h1>
<h2>Purpose:</h2>
<ul>
<li>Fix current behavior to match the acceptance criteria</li>
</ul>
<h2>Approach</h2>
<ul><li>Code refactoring</li>
<h2>ToDos</h2>
<ul>
<li>Hide Opening time offset field on details pane if the Closed library due date management field value differs from 'Move to the beginning of the next open service point hours'</li>
<li>Set a default units for the Opening time offset field to Hours</li>
<li>Allow only numbers to be set in `PolicyPropertySetter` input field</li>
<li>Add minimum edge to the inputs</li>
</ul>
</ul>
<h2>Bug links</h2>
<a href="https://issues.folio.org/browse/UICIRC-113">https://issues.folio.org/browse/UICIRC-113</>

### CODE REVIEW

File: settings/LoanPolicyDetail.js [hunk 1]
```diff
@@ -101,6 +102,16 @@ class LoanPolicyDetail extends React.Component {
     const exReqPerInterval = get(policy, ['loansPolicy', 'existingRequestsPeriod', 'intervalId']);
     const gracePeriodInterval = get(policy, ['loansPolicy', 'gracePeriod', 'intervalId']);
     const timeOffsetInterval = get(policy, ['loansPolicy', 'openingTimeOffset', 'intervalId']);
+    const isOpeningTimeOffsetVisible = () => {
+      const {
+        loansPolicy,
```
Comment (maximdidenkoepam) [inline settings/LoanPolicyDetail.js]: Are you sure about this variable? As far I remember it should be shortTermLoan and selected item from there. In our case it is `BEGINNING_OF_THE_NEXT_OPEN_SERVICE_POINT_HOURS`

File: settings/LoanPolicyForm.js [hunk 2]
```diff
@@ -122,10 +122,14 @@ class LoanPolicyForm extends React.Component {
 
     const openingTimeOffset = (allValues.loansPolicy && allValues.loansPolicy.openingTimeOffset);
     const openingTimeOffsetIsVisible = openingTimeOffset && openingTimeOffset.intervalId && !openingTimeOffset.duration;
+    const openingTimeOffsetIsDefault = openingTimeOffset && openingTimeOffset.duration && !openingTimeOffset.intervalId;
```
Comment (maximdidenkoepam) [inline settings/LoanPolicyForm.js]: Explain why do we need it and to if statements with `this.props.change`?

Comment (anton-plotnikov) [reply settings/LoanPolicyForm.js]: The default units for this field are 'Hours', that is why when we select a value it automatically measures it in  'Hours'

File: settings/LoanPolicyDetail.js (hunk 1 above)
Comment (anton-plotnikov) [reply settings/LoanPolicyDetail.js]: Definitely it should be shortTermLoan, but openingTimeOffset field appears only when `BEGINNING_OF_THE_NEXT_OPEN_SERVICE_POINT_HOURS` is selected in shortTerms

Comment (maximdidenkoepam) [reply settings/LoanPolicyDetail.js]: and what if tomorrow this value will be available for long term load too?

Comment (anton-plotnikov) [reply settings/LoanPolicyDetail.js]: got the point

File: settings/LoanPolicyForm.js (hunk 2 above)
Comment (anton-plotnikov) [reply settings/LoanPolicyForm.js]: Sorry, but do you have any concerns still?

File: settings/LoanPolicyDetail.js [hunk 3]
```diff
@@ -153,15 +173,19 @@ class LoanPolicyDetail extends React.Component {
...
-        <Row>
-          <Col xs={12}>
-            <KeyValue
-              label={<FormattedMessage id="ui-circulation.settings.loanPolicy.openingTimeOffset" />}
-              value={`${get(policy, ['loansPolicy', 'openingTimeOffset', 'duration'], '')} ${timeOffsetInterval}`}
-            />
-          </Col>
-        </Row>
-        <br />
+        {(isOpeningTimeOffsetVisible()) &&
```
Comment (maximdidenkoepam) [inline settings/LoanPolicyDetail.js]: `this.isOpeningTimeOffsetVisible()` and I suggest to move it to a separate variable

**Pull request merged**